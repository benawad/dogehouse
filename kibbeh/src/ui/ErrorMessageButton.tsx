import React from "react";

export const ErrorMessageButton: React.FC<
  React.ComponentPropsWithoutRef<"button">
> = ({ children, className, style, ...props }) => {
  return (
    <button
      className={`rounded-lg px-3 font-bold text-sm bg-secondary-washed-out ${className}`}
      data-testid="error-msg-btn"
      style={{
        paddingTop: 3,
        paddingBottom: 3,
        ...style,
      }}
      {...props}
    >
      {children}
    </button>
  );
};
