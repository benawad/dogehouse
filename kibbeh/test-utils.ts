import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
