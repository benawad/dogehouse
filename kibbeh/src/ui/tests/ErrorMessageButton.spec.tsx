import React from "react";

import { render } from "../../../test-utils";
import { ErrorMessageButton } from "../ErrorMessageButton";

describe("ErrorMessageButton", () => {
  it("should render and input correctly", () => {
    const { getByTestId } = render(<ErrorMessageButton />);
    const component = getByTestId("error-msg-btn");

    expect(component).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<ErrorMessageButton />);
    const component = getByTestId("error-msg-btn");
    expect(component).toMatchSnapshot();
  });
});
