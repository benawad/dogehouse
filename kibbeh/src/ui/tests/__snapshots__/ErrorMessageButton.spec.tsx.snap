// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessageButton should match snapshot 1`] = `
<button
  class="rounded-lg px-3 font-bold text-sm bg-secondary-washed-out undefined"
  data-testid="error-msg-btn"
  style="padding-top: 3px; padding-bottom: 3px;"
/>
`;
